src/main/java/net/imagej/ops/filter/derivative/PartialDerivativeRAI.java

```diff
@@ -0,0 +1,149 @@
+package net.imagej.ops.filter.derivative;
+
+import net.imagej.ops.Ops;
+import net.imagej.ops.Ops.Filter.PartialDerivative;
+import net.imagej.ops.special.chain.RAIs;
+import net.imagej.ops.special.computer.BinaryComputerOp;
+import net.imagej.ops.special.computer.UnaryComputerOp;
+import net.imagej.ops.special.function.UnaryFunctionOp;
+import net.imagej.ops.special.hybrid.AbstractUnaryHybridCF;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.img.Img;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.type.numeric.real.DoubleType;
+import net.imglib2.view.IntervalView;
+import net.imglib2.view.MixedTransformView;
+import net.imglib2.view.Views;
+
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+/**
+ * Calculates the derivative (with sobel kernel) of an image in a given
+ * dimension.
+ * 
+ * @author Eike Heinz, University of Konstanz
+ *
+ * @param <T>
+ *            type of input
+ */
+@Plugin(type = Ops.Filter.PartialDerivative.class, name = Ops.Filter.PartialDerivative.NAME)
+public class PartialDerivativeRAI<T extends RealType<T>> extends
+		AbstractUnaryHybridCF<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>> implements PartialDerivative {
+
+	@Parameter
+	private int dimension;
+
+	private UnaryFunctionOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>> createRAI;
+
+	private BinaryComputerOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>, RandomAccessibleInterval<T>> addOp;
+
+	private UnaryComputerOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>> kernelBConvolver;
+
+	private UnaryComputerOp<RandomAccessibleInterval<T>, RandomAccessibleInterval<T>>[] kernelAConvolverArray;
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public void initialize() {
+		RandomAccessibleInterval<T> kernel = ops().create().kernelSobelSeparated();
+		// kernel A contains 1 2 1
+		IntervalView<T> kernelA = Views.hyperSlice(Views.hyperSlice(kernel, 3, 0), 2, 0);
+		// kernel B contains -1 0 1
+		IntervalView<T> kernelB = Views.hyperSlice(Views.hyperSlice(kernel, 3, 0), 2, 1);
+
+		// add dimensions to kernel if input has more than 2 dimensions to
+		// properly rotate the kernel
+		if (in().numDimensions() > 2) {
+			MixedTransformView<T> expandedKernelA = Views.addDimension(kernelA);
+			MixedTransformView<T> expandedKernelB = Views.addDimension(kernelB);
+			for (int i = 0; i < in().numDimensions() - 3; i++) {
+				expandedKernelA = Views.addDimension(expandedKernelA);
+				expandedKernelB = Views.addDimension(expandedKernelB);
+			}
+			long[] dims = new long[in().numDimensions()];
+			for (int j = 0; j < in().numDimensions(); j++) {
+				dims[j] = 1;
+			}
+			dims[0] = 3;
+			Img<DoubleType> kernelInterval = ops().create().img(dims);
```

Comment:
You don't seem to be using the created image? Replace that line with `Interval kernelInterval = new FinalInterval(dims);`